### PR TITLE
implement notion block types as classes

### PR DESCRIPTION
**Pull request by johguentner**
- add easy content and text access for rich-text- (all supported blocks, exept page-blocks) and child-page-blocks
- add fromResponse- and mapTypeToClass-method to Block::class (similar to the implementations within Property::class)

### CODE REVIEW

**Review by mechelon — CHANGES_REQUESTED** [review]

File: src/Entities/Blocks/Block.php [hunk 1]
```diff
@@ -133,4 +144,61 @@ public function getLastEditedTime(): DateTime
...
+            case 'bulleted_list_item':
+            case 'numbered_list_item':
+            case 'child_page':
+            case 'paragraph':
+            case 'to_do':
+            case 'toggle':
+                $class = str_replace('_', '', ucwords($type, '_'));
+                return "FiveamCode\\LaravelNotionApi\\Entities\\Blocks\\" . $class;
+            case 'heading_1':
+                return "FiveamCode\\LaravelNotionApi\\Entities\\Blocks\\HeadingOne";
```
[the remark this answers is not in the document]
Comment (johguentner) [reply src/Entities/Blocks/Block.php]: ✅ done

File: src/Entities/Collections/BlockCollection.php [hunk 2]
```diff
@@ -12,11 +12,40 @@
  */
 class BlockCollection extends EntityCollection
 {
+    private bool $showUnsupported = false;
+
+    public function withUnsupported(): BlockCollection
```
[the remark this answers is not in the document]
Comment (johguentner) [reply src/Entities/Collections/BlockCollection.php:24]: ✅ done

File: src/Entities/Collections/BlockCollection.php [hunk 3]
```diff
@@ -12,11 +12,40 @@
...
-        foreach ($this->rawResults as $blockChild) {
-            $this->collection->add(new Block($blockChild));
+        foreach ($this->rawResults as $blockChildContent) {
+            $this->collection->add(Block::fromResponse($blockChildContent));
+        }
+    }
+
+    public function asCollection(): Collection
+    {
+        $collection =parent::asCollection();
```
[the remark this answers is not in the document]
Comment (johguentner) [reply src/Entities/Collections/BlockCollection.php]: ✅ done

File: src/Entities/Collections/BlockCollection.php [hunk 4]
```diff
@@ -12,11 +12,40 @@
...
+        if ($this->showUnsupported) {
+            return $collection;
+        } else {
+            return $collection->filter(function($block){
+                return $block->getType() != 'unsupported';
```
[the remark this answers is not in the document]
Comment (johguentner) [reply src/Entities/Collections/BlockCollection.php]: ✅ done

**Review by johguentner — COMMENTED** [review]
applied all necessary changes

**Review by johguentner — COMMENTED** [review]
checked

**Pull request merged**